kmip/tests/core/secrets/test_certificate.py

```diff
@@ -0,0 +1,378 @@
+# Copyright (c) 2015 The Johns Hopkins University/Applied Physics Laboratory
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from testtools import TestCase
+
+from kmip.core.attributes import CertificateType
+from kmip.core.enums import CertificateTypeEnum
+from kmip.core.misc import CertificateValue
+from kmip.core.secrets import Certificate
+from kmip.core.utils import BytearrayStream
+
+
+class TestCertificate(TestCase):
+    """
+    A test suite for the Certificate class.
+    """
+
+    def setUp(self):
+        super(TestCertificate, self).setUp()
+
+        self.certificate_type_a = None
+        self.certificate_type_b = CertificateTypeEnum.X_509
+
+        self.certificate_value_a = None
+        self.certificate_value_b = (
```

Comment:
Could you add a comment saying how these bytes were generated or where they came from.